zerver/lib/actions.py

```diff
@@ -1011,6 +1011,10 @@ def check_message(sender, client, message_type_name, message_to,
     message.recipient = recipient
     if message_type_name == 'stream':
         message.subject = subject
+
+    message.rendered_content = message.render_markdown(message_content)
+    message.is_me_message = Message.is_status_message(message_content, message.rendered_content)
```

Comment:
Yeah, I think if you move your `is_status_message` check after the `message.maybe_render_content(realm.domain)` call a few lines down, you won't need to add the redundant render call here (and it might also fix the issue you had below with sending_client, since it is set in the intervening lines).